pkg/products/threescale/prometheusRules.go

```diff
@@ -329,8 +333,78 @@ func (r *Reconciler) newAlertReconciler(logger l.Logger, installType string, ctx
+				GroupName: fmt.Sprintf("%s-general.rules", installationName),
+				Rules: []monitoringv1.Rule{
+					{
+						Alert: fmt.Sprintf("%sThreescaleCriticalMetricsMissing", strings.ToUpper(installationName)),
+						Annotations: map[string]string{
+							"sop_url": resources.SopUrlCriticalMetricsMissing,
+							"message": "one or more critical metrics have been missing for 10+ minutes",
+						},
+						Expr:   missingMetricsExpr,
+						For:    "15m",
+						Labels: map[string]string{"severity": "critical"},
+					},
+				},
+			},
 		},
-	}, nil
+	}
+
+	if customDomain.IsCustomDomain(r.installation) {
+		alerts.Alerts = append(alerts.Alerts,
+			resources.AlertConfiguration{
+				AlertName: fmt.Sprintf("%s-custom-domain-alert", installationName),
+				Namespace: namespace,
+				GroupName: fmt.Sprintf("%s-custom-domaim.rules", installationName),
+				Rules: []monitoringv1.Rule{
+					{
+						Alert: "CustomDomainCRErrorState",
+						Annotations: map[string]string{
+							"sop_url": resources.SopUrlRHOAMServiceDefinition,
+							"message": "Error configuring custom domain, please refer to the documentation to resolve the error.",
+						},
+						Expr:   intstr.FromString(fmt.Sprintf("%s_custom_domain{active='true'} > 0", installationName)),
+						For:    "5m",
+						Labels: map[string]string{"severity": "warning", "product": installationName},
+					},
+					{
+						Alert: "DnsBypassThreeScaleAdminUI",
+						Annotations: map[string]string{
+							"sop_url": resources.SopUrlDnsBypassThreeScaleAdminUI,
+							"message": "3Scale Admin UI, bypassing DNS: If this console is unavailable, the client is unable to configure or administer their API setup.",
+						},
+						Expr:   intstr.FromString("threescale_portals{system_master='false'} > 0"),
+						For:    "15m",
+						Labels: map[string]string{"severity": "critical", "product": installationName},
+					},
+					{
+						Alert: "DnsBypassThreeScaleDeveloperUI",
+						Annotations: map[string]string{
+							"sop_url": resources.SopUrlDnsBypassThreeScaleDeveloperUI,
+							"message": "3Scale Developer UI, bypassing DNS: If this console is unavailable, the client developers are unable signup or perform API management.",
+						},
+						Expr:   intstr.FromString("threescale_portals{system_developer='false'} > 0"),
+						For:    "15m",
+						Labels: map[string]string{"severity": "critical", "product": installationName},
+					},
+					{
+						Alert: "DnsBypassThreeScaleSystemAdminUI",
+						Annotations: map[string]string{
+							"sop_url": resources.SopUrlDnsBypassThreeScaleSystemAdminUI,
+							"message": "3Scale System Admin UI, bypassing DNS: If this console is unavailable, the client is unable to perform Account Management, Analytics or Billing.",
+						},
+						Expr:   intstr.FromString("threescale_portals{system_provider='false'} > 0"),
+						For:    "15m",
+						Labels: map[string]string{"severity": "critical", "product": installationName},
+					},
```

Comment:
The DNS Bypass alerts also apply to clusters that are on a private network without public routes. These need to be part of every install.

Reply:
ack. I'll update that. 👍

Reply:
updated